### PR TITLE
[azure] auto case OCP-24152 OCP-24203 OCP-49501

### PR DESCRIPTION
**Pull request by jinyunma**
auto below test cases:
OCP-49501 Validate UltraSSD disks on control plane and compute instances
OCP-24203 [ipi-on-azure] Create machines in multiple AZs
OCP-24152 [azure] [CORS-1102](https://issues.redhat.com//browse/CORS-1102) - Confirm that master and worker nodes are running Red Hat Enterprise Linux CoreOS

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ workflow:
   steps: 
     pre:
     - chain: cucushift-installer-rehearse-azure-ipi-provision
+    - chain: cucushift-installer-check-azure
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml:6]: This chain can be extended to all the install workflow?

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml:6]: yeah, those chain can be run on most azure ipi workflows. But there is a rate to hit below prow system error on step, in order to make less noise for CI job, how about to only run this chain in some workflows?

```
INFO[2023-10-27T17:58:18Z] Running step aws-ipi-sdn-mtu-migrate-f28-network-mtu-migration-decrease-end. 
INFO[2023-10-27T18:28:18Z] pod pending for more than 30m0s: containers have not started in 30m0.000147894s: ci-scheduling-dns-wait, cp-entrypoint-wrapper, place-entrypoint, sidecar, test: 
* Container sidecar is not ready with reason PodInitializing
* Container test is not ready with reason PodInitializing 
```

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml:6]: >  how about to only run this chain in some workflows

Understood, okay for me.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml:6]: Beside `cucushift-installer-rehearse-azure-ipi-usertags`, are we going to extend it to some other install workflow, or in our current plan, this is the only one.

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml:6]: yeah, I just added chain into install chains `cucushift-installer-rehearse-azure-ipi-sdn` and `cucushift-installer-rehearse-azure-mag-ipi` to see if it works well on those workflows.

File: ci-operator/step-registry/cucushift/installer/check/azure/coreos/cucushift-installer-check-azure-coreos-commands.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+no_critical_check_result=0
+echo "------ Checking that all ndoes are running Red Hat Enterprise Linux CoreOS ------"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/azure/coreos/cucushift-installer-check-azure-coreos-commands.sh:50]: seem like we can prompt this checking to higher level, not limited to a specific platform, and we also can remove the above az lines.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Pull request merged**